mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -557,6 +604,180 @@ class WidgetAdapter(
         }
     }
 
+    class InputViewHolder internal constructor(initData: ViewHolderInitData) :
+        LabeledItemBaseViewHolder(initData, R.layout.widgetlist_inputitem, R.layout.widgetlist_inputitem_compact) {
+        private val inputTextLayout: TextInputLayout = itemView.findViewById(R.id.widgetinput)
+        private val inputText: TextInputEditText = itemView.findViewById(R.id.widgetinputvalue)
+        private var isBinding = false
+        private var hasChanged = false
+        private var showingDatePicker = false
+        private var showingTimePicker = false
+
+        private var updateJob: Job? = null
+        private var oldValue: String? = null
+
+        init {
+            inputText.doAfterTextChanged {
+                if (!isBinding) {
+                    hasChanged = true
+                }
+            }
+            inputText.setOnFocusChangeListener { _, hasFocus ->
+                if (!hasFocus && hasChanged) {
+                    updateValue()
+                }
+            }
```

Comment:
BasicUI gets away without the need for an extra update button. And I would prefer that. Also notice that it is not relevant for date/time, that have their own buttons on the popup. For me, the change on focus didn't give me a problem (apart from the keyboard not being close automatically, which I think it should be possible to resolve).

Reply:
> BasicUI gets away without the need for an extra update button

Well, yes, it's certainly *possible*. I'm just not sure whether it's *good UX*. The button would automatically give you a 'hook' for hinding the keyboard. I don't see how it would be possible to auto-hide the keyboard otherwise. Maybe [this API](https://developer.android.com/reference/android/widget/TextView.html#setOnEditorActionListener%28android.widget.TextView.OnEditorActionListener%29), handling keyboard 'Enter' key press, could be a compromise.

> Also notice that it is not relevant for date/time, that have their own buttons on the popup

I am aware, it's why I suggested the button: that way, *all* inputs would have an explicit way of saving.

Reply:
The keyboard also has a return button, so I would prefer that. Looking at Google apps, I don't see them having an explicit confirm button, only on a popup or overlay page with multiple input elements. But I am OK to go the other way as well.